homeassistant/components/moonraker/sensor.py

```diff
@@ -0,0 +1,171 @@
+        entity_registry_enabled_default=True,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:printer-3d-nozzle",
+    ),
+    MoonrakerSensorDescription(
+        key="extruder_target",
+        name="Extruder Target Temperature",
+        signal=SIGNAL_UPDATE_EXTRUDER,
+        value=lambda params: round(params["target"], 1),
+        entity_registry_enabled_default=True,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:printer-3d-nozzle",
+    ),
+    MoonrakerSensorDescription(
+        key="bed_temperature",
+        name="Bed Temperature",
+        signal=SIGNAL_UPDATE_HEAT_BED,
+        value=lambda params: round(params["temperature"], 1),
+        entity_registry_enabled_default=True,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:radiator",
+    ),
+    MoonrakerSensorDescription(
+        key="bed_target",
+        name="Bed Target Temperature",
+        signal=SIGNAL_UPDATE_HEAT_BED,
+        value=lambda params: round(params["target"], 1),
+        entity_registry_enabled_default=True,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        icon="mdi:radiator",
+    ),
+    MoonrakerSensorDescription(
+        key="print_progress",
+        name="Print Progress",
+        signal=SIGNAL_UPDATE_VIRTUAL_SDCARD,
+        value=lambda params: round(params["progress"] * 100),
+        entity_registry_enabled_default=True,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+    MoonrakerSensorDescription(
+        key="print_duration",
+        name="Print Duration",
+        signal=SIGNAL_UPDATE_PRINT_STATUS,
+        value=lambda params: str(
+            datetime.timedelta(seconds=round(params["print_duration"]))
+        ),
+        entity_registry_enabled_default=True,
+        icon="mdi:clock-outline",
+    ),
+    MoonrakerSensorDescription(
+        key="print_file",
+        name="Print File",
+        signal=SIGNAL_UPDATE_PRINT_STATUS,
+        value=lambda params: params["filename"],
+        entity_registry_enabled_default=True,
+        icon="mdi:file",
+    ),
+)
+
+
+class MoonrakerGenericSensor(MoonrakerEntity, SensorEntity):
+    """Binary sensor representing printing state."""
+
+    entity_description: MoonrakerSensorDescription
+
+    def __init__(
+        self,
+        entry: ConfigEntry,
+        connector: APIConnector,
+        description: MoonrakerSensorDescription,
+    ) -> None:
+        """Initialize a new printing binary sensor."""
+        super().__init__(entry, connector, description.name)
```

Comment:
We do not need to pass the name, its picked up from the entity description by default
```suggestion
        super().__init__(entry, connector)
```